test_program/block_test.c

```diff
@@ -0,0 +1,60 @@
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <inttypes.h>
+#include <cmockery/cmockery.h>
+extern int get_block(int pos);
+extern int init_block_store();
+extern int insert_block(char *buffer,size_t length);
+extern int fini_block_store();
+static void test_return_values(void **state) 
+{ 
+        
+        assert_int_equal(init_block_store(), 0);
+
+}
+
+/* Test case that fails as leak_memory() leaks a dynamically allocated block.*/
+static void leak_memory_test(void **state) 
+{
+        
+        get_block(5);
```

Comment:
Ewen/Srinivas can we have this fixed asap. If u have a doubt please ask Luis about it.
Also this patch is old and will have merge conflicts so could u please fix this on a new patch.